pontoon/base/views.py

```diff
@@ -294,19 +304,27 @@ def get_translations_from_other_locales(request):
         }, status=400)
 
     entity = get_object_or_404(Entity, pk=entity)
-    locales = entity.resource.project.locales.exclude(code=locale)
+    locale = get_object_or_404(Locale, code=locale)
+    preferred_locales = request.user.profile.sorted_locales
     plural_form = None if entity.string_plural == "" else 0
 
     translations = Translation.objects.filter(
         entity=entity,
-        locale__in=locales,
         plural_form=plural_form,
         approved=True
-    ).order_by('locale__name')
+    ).exclude(locale=locale)
+
+    preferred = translations.filter(locale__in=preferred_locales)
+    other = translations.exclude(locale__in=preferred_locales)
+
+    preferred_locales_list = list(_get_translation_values(preferred))
```

Comment:
We should keep the order of locales the same. Taking into account the suggested naming changes, you can do it like this:
```python
preferred_translations = sorted(
    _get_translation_values(preferred),
    key=lambda t: request.user.profile.locales_order.index(t['locale_pk'])
)
```